pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -432,14 +432,16 @@ stdenv.mkDerivation {
     # Always add -march based on cpu in triple. Sometimes there is a
     # discrepency (x86_64 vs. x86-64), so we provide an "arch" arg in
     # that case.
-    + optionalString ((targetPlatform ? gcc.arch) &&
+    # TODO: aarch64-darwin has mcpu incompatible with gcc
+    + optionalString ((targetPlatform ? gcc.arch) && (isClang || !(stdenv.isDarwin && stdenv.isAarch64)) &&
```

Comment:
What exactly is the problem here?

Reply:
#105026 sets CPU to something GCC doesn't know about: 

* https://github.com/NixOS/nixpkgs/pull/105026/files#r587172323

Reply:
Thanks